### PR TITLE
✅ Test news file discovery

### PR DESCRIPTION
**Pull request by muchzill4**
- extracted all the git work to `GitWrapper` (agreed with @acabarbaye that it's a good thing to do)
- had to change the code a bit, because it was really hard to test with state creeping in
- sneaked in docstrings removal, as we no longer need them

Really struggling with file naming pattern here. I decided to go with describing what functionality they test versus concrete class/function name.

### CODE REVIEW

File: mbed_tools_ci/utils/git_helpers.py [hunk 1]
```diff
@@ -439,3 +439,14 @@ def _get_remote(self) -> Optional[Any]:
...
             return None
+
+    def get_changes_against_master(self) -> List[str]:
+        """Returns a list of files changed against master branch."""
+        master_branch_commit = self.repo.commit('master')
+        current_branch_commit = self.repo.commit(self.get_current_branch())
+        changes = self.get_changes_list(
+            self.get_branch_point(master_branch_commit, current_branch_commit),
+            current_branch_commit,
+            change_type='a'
```
Comment (acabarbaye) [inline mbed_tools_ci/utils/git_helpers.py:454]: this would list only `added` files. Could we either mention this in the function name or let it be an argument of the method

File: mbed_tools_ci/assert_news.py [hunk 2]
```diff
@@ -42,133 +40,64 @@ def validate_file_contents(self) -> None:
...
     try:
-        NewsFileDiscoverer().verify()
+        validate_news_files(git=GitWrapper(), news_dir=str(relative_news_dir))
     except Exception as e:
         log_exception(logger, e)
         sys.exit(1)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
```
Comment (acabarbaye) [inline mbed_tools_ci/assert_news.py:137]: Has this file been "blacked"?

File: mbed_tools_ci/assert_news.py [hunk 3]
```diff
@@ -42,133 +40,64 @@ def validate_file_contents(self) -> None:
...
 def main() -> None:
     """Asserts the new PR comprises at least one news file and it adheres to the required standard.
 
     An exception is raised if a problem with the news file is found.
     """
-    parser = argparse.ArgumentParser(
-        description='Publish target data report to AWS.')
-    parser.add_argument("-v", "--verbose", action="count", default=0,
-                        help="Verbosity, by default errors are reported.")
+    parser = argparse.ArgumentParser(description="Publish target data report to AWS.")
```
Comment (acabarbaye) [inline mbed_tools_ci/assert_news.py]: Just realised the description of the CLI is incorrect

File: mbed_tools_ci/assert_news.py [hunk 4]
```diff
@@ -42,133 +40,64 @@ def validate_file_contents(self) -> None:
...
-        project_root = configuration.get_value(
-            ConfigurationVariable.PROJECT_ROOT)
-        if not added_news or len(added_news) == 0:
-            raise FileNotFoundError(
-                f'PR must contain a news file in {news_dir}. See README.md'
-            )
-        logger.info(
-            f'{len(added_news)} new news files found in `{news_dir}`'
+def validate_news_files(git: GitWrapper, news_dir: str) -> None:
+    """Checks that news files exist and pass validation checks."""
```
Comment (acabarbaye) [inline mbed_tools_ci/assert_news.py]: Could you document what the function parameters mean?
e.g. it is important to say than news_dir must be the relative path from project root

**Review by acabarbaye — CHANGES_REQUESTED** [review]

**Comment by acabarbaye** [comment]
I have some minor comments.
Looks great and much clearer than it was

File: mbed_tools_ci/assert_news.py (hunk 3 above)
Comment (muchzill4) [reply mbed_tools_ci/assert_news.py]: Ha. It is!

File: mbed_tools_ci/assert_news.py (hunk 2 above)
Comment (muchzill4) [reply mbed_tools_ci/assert_news.py:137]: Don't know what you're talking about. 😉 Is it a problem? I'd probably make quotes consistent manually if it wasn't `black`.

File: mbed_tools_ci/utils/git_helpers.py (hunk 1 above)
Comment (muchzill4) [reply mbed_tools_ci/utils/git_helpers.py:454]: Good point, I'll change the name.

File: mbed_tools_ci/assert_news.py (hunk 4 above)
Comment (muchzill4) [reply mbed_tools_ci/assert_news.py]: Nice catch. Either that or I make it work on absolutes, all the way. What do you think? It'd require some additional logic resolving changes from git properly. 🤔

**Comment by muchzill4** [comment]
@acabarbaye I think I addressed all your comments. Do tell.

File: tests/assert_news/test_news_file_validation.py [hunk 5]
```diff
@@ -1,20 +1,17 @@
...
 
 
 class TestValidateFileName(TestCase):
-    """Tests for the NewsFileValidator.validate_file_name method."""
-
     def test_raises_if_file_name_is_not_valid(self):
-        """Given an invalid file name, it raises ValueError."""
         with self.assertRaises(ValueError) as cm:
-            NewsFileValidator("imaginary/path/to/file.tar.gz").validate_file_name()
+            NewsFileValidator("/imaginary/path/to/file.tar.gz").validate_file_name()
```
Comment (rosiemiller2) [inline tests/assert_news/test_news_file_validation.py:14]: Just out of interest - why did you change this?

Comment (muchzill4) [reply tests/assert_news/test_news_file_validation.py:14]: It was to make the absoluteness of paths more explicit (in code and tests). They are not strictly necessary, provided you're in good directory and news files are available relatively to where you are.

There's some hairy stuff going on with relative vs absolute paths in this code anyway. `configuration` returns absolute paths always (which is good), but `GitWrapper` returns changes relative to project root. I decided to use `absolute` paths always for clarity.

File: mbed_tools_ci/assert_news.py [hunk 6]
```diff
@@ -20,136 +18,86 @@
...
-        added_news = self.git.get_changes_list(
-            self.git.get_branch_point(
-                master_branch_commit, current_commit),
-            current_commit, change_type='a',
-            dir=news_dir_relative_path
-        )
-        extension_to_exclude = ['.toml', '.rst']
-        return [path for path in added_news if
-                len([ex for ex in extension_to_exclude if ex in path]) == 0]
+def validate_news_file(absolute_path: str) -> None:
```
Comment (rosiemiller2) [inline mbed_tools_ci/assert_news.py:55]: I much prefer not using classes if we don't have to 👍 I find it difficult to test code where I have to create full instances of classes just to test small internal methods.

**Review by rosiemiller2 — APPROVED** [review]

File: tests/assert_news/test_news_file_validation.py (hunk 5 above)
Comment (rosiemiller2) [reply tests/assert_news/test_news_file_validation.py:14]: Hmm yeah its something I haven't been very aware of so far. Will keep it in mind for any future stuff.

**Review by acabarbaye — APPROVED** [review]
